sdk/storage/azblob/internal/exported/set_expiry.go

```diff
@@ -7,16 +7,18 @@
 package exported
 
 import (
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
-	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/generated"
 	"net/http"
 	"strconv"
 	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/internal/generated"
 )
 
 // ExpiryType defines values for ExpiryType
 type ExpiryType interface {
 	Format(o *SetExpiryOptions) (generated.ExpiryOptions, *generated.BlobClientSetExpiryOptions)
+	notPubliclyImplementable()
```

Comment:
This is part of "internal" package so this will not be exposed outside for public use anyway. Is there any problem you see with the current implementation?

Reply:
The convention is that types in the `internal/exported` package are exported as public surface area (unless comments indicate otherwise).  In this case, `ExpiryType` is exported from the `appendblob` and `blockblob` packages.